### ISSUE TITLE
False Positive on Layout/SpaceInsideParens Cop

### ISSUE THREAD

**Issue by calebtocco** [sds/haml-lint#625]
## Problem

When using multi-line tag attributes with the common HAML style of spaces inside braces (`{ foo: bar }`), haml-lint reports `Layout/SpaceInsideParens` violations that cannot be fixed in the source HAML.

## Environment

- haml-lint: 0.69.0
- rubocop: 1.82.0
- Ruby: 3.4.7
- Rails: 7.2.3

## Steps to Reproduce

Create a HAML file with multi-line attributes:

```haml
%a.nav-link{ href: '#tab-foo',
             data: { action: 'click->nav#activate' },
             role: 'tab' }
```

Run haml-lint:

```
$ haml-lint app/views/example.html.haml
app/views/example.html.haml:1 [W] RuboCop: Layout/SpaceInsideParens: Space inside parentheses detected.
```

## Root Cause

The `ChunkExtractor#wrap_lines` method wraps extracted attribute content in parentheses for RuboCop processing. When HAML has `{ href:` (space after `{`) or `'tab' }` (space before `}`), the extracted Ruby becomes `W( href:...` or `...'tab' )`, triggering `Layout/SpaceInsideParens`.

This is a style mismatch: HAML commonly uses `{ foo: bar }` with spaces, but RuboCop's `Layout/SpaceInsideParens` doesn't allow spaces after `(` or before `)`.

## Proposed Solution

Modify `wrap_lines` to strip leading/trailing whitespace when **linting** (to avoid false positives), but preserve it when **autocorrecting** (so RuboCop can fix it and map corrections back correctly).

**chunk_extractor.rb:**

```ruby
def wrap_lines(lines, wrap_depth)
  lines = lines.dup
  wrapping_prefix = 'W' * (wrap_depth - 1) + '('

  # Strip leading/trailing space only when linting (not autocorrecting) to avoid
  # SpaceInsideParens violations. During autocorrect, preserve spaces so RuboCop
  # can fix them and map corrections back correctly.
  if autocorrect
    lines[0] = wrapping_prefix + lines[0]
    lines[-1] = lines[-1] + ')'
  else
    leading_spaces = lines[0][/^\s*/].length
    wrapping_prefix = 'W' * (wrap_depth - 1 + leading_spaces) + '('
    lines[0] = wrapping_prefix + lines[0].lstrip
    lines[-1] = lines[-1].rstrip + ')'
  end

  lines
end
```

This requires passing an `autocorrect` flag through:
- `Linter::RuboCop` → `Coordinator` → `ChunkExtractor`

I have a working implementation with passing tests that I can provide as a PR if desired.

**Comment by sds** [comment]
Thanks!
